### ISSUE TITLE
[1.14.4-PE1.0.1B] Middle click in energy condenser inventory wipes out inventory

### ISSUE THREAD

**Issue by jmbjorndalen** [sinkillerj/ProjectE#1953]
**Tickets that do not conform to this template will be closed without comment**

Exact ProjectE version (do not say "latest", "latest on Curse", or similar): 
ProjectE-1.14.4-PE1.0.1B.jar   (downloaded from curseforge, but also observed with 0B)

Exact Forge version:
1.14.4-28.1.113  (but also observed with earlier Forge versions)

Link to crash log (please use a paste site, do not attach the .txt or paste the log inline): 
No crash. 

Steps to reproduce:
1. Let energy condenser (or energy condenser mk2) produce multiple stacks of an item. 
2. Use middle mouse button to sort inventory (click in inventory). 

What I expected to happen:
Sorted inventory. 

What happened instead:
Empty inventory. 

Additional info: 

I have added ProjectE to All the Mods 4 (currently version 0.3.1). I assume that "Inventory Sorter" is the mod that is trying to run the sort operation (inventorysorter-1.14.4-16.2.0.jar) based on key bindings in options. 

Using the "sort" button on the inventory does nothing to the condenser inventory. On the positive side, it doesn't wipe the inventory.

**Comment by jmbjorndalen** [comment]
Thinking more about this, the issue is probably with Inventory Sorter. I tried replacing it with Inventory Tweaks Reloaded (invtweaks-1.14.4-0.5.0.jar). This failed in a more sensible way: refusing to sort the inventory. 

I'm only guessing, but the energy condensers normally refuse to insert items of the same type as the ones generated. If inventory sorter tries to remove and re-insert items to sort a chest/inventory, it will fail to do so with the condensers (remove should work, insert will fail). 

Checking the debug log, I find several entries like these ones: 

> [10Jan2020 17:33:59.054] [Server thread/DEBUG] [cpw.mods.inventorysorter.InventorySorter/]: Item 64 diamond is not valid in slot 91 of container inventorysorter:dummyplayercontainer


The debug messages match the source code in
https://github.com/cpw/inventorysorter/blob/master/src/main/java/cpw/mods/inventorysorter/SortingHandler.java

I have found a couple of related issues in case others run into the same problem. 

https://github.com/cpw/inventorysorter/issues/58

https://github.com/cpw/inventorysorter/issues/45

The solution appears to be to add condensers to a blacklist. 

I think this issue can be closed, but I'm not familiar enough with Forge code to see if there is something the condensers should do to "warn" the inventory sorter. From my very rough understanding, the inventory sorter code seems vulnerable to 1) race conditions (others fill inventory concurrently with sorting) and 2) inventories that refuse to take back items. I assume that the first issue doesn't happen as the code is single threaded and this code (apparently) doesn't allow any other inventory updates to happen while it is running. 

Thank you, and sorry about the noise.
